File: Programmers/Lv0/평행.py
def solution(dots):
    inc = []
    for i in range(len(dots) - 1):
        for j in range(i + 1, len(dots)):
            if (dots[i][1] - dots[j][1] == 0 and dots[i][0] - dots[j][0] == 0):
                return 1
            if dots[j][0] - dots[i][0] == 0:
                inc.append((dots[i][1] - dots[j][1]) / (dots[i][0] - dots[j][0]))
            else:
                inc.append((dots[j][1] - dots[i][1]) / (dots[j][0] - dots[i][0]))
    for k in range(len(inc) // 2):
        if inc[k] == inc[len(inc) - k - 1]:
            return 1
    return 0

File: Programmers/Lv0/평행.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> dots) {
    vector<double> inc;
    for (size_t i = 0; i < dots.size() - 1; i++) {
        for (size_t j = i + 1; j < dots.size(); j++) {
            if ((dots[i][1] - dots[j][1]) == 0 && (dots[i][0] - dots[j][0]) == 0)
                return 1;
            if ((dots[j][0] - dots[i][0]) == 0)
                inc.push_back((dots[i][1] - dots[j][1]) / double(dots[i][0] - dots[j][0]));
            else
                inc.push_back((dots[j][1] - dots[i][1]) / double(dots[j][0] - dots[i][0]));
        }
    }
    for (int i = 0; i < inc.size() / 2; i++)
        if (inc[i] == inc[inc.size() - i - 1])
            return 1;
    return 0;
}